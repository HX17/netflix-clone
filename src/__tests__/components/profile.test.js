import React from "react";
import { render } from "@testing-library/react";
import Profile from "../../components/profile";

describe("<Profile />", () => {
  it("renders the <Profile /> with populated data", () => {
    const { container, getByTestId, getByText } = render(
      <Profile>
        <Profile.Title>Who's watching?</Profile.Title>
        <Profile.List>
          <Profile.User onClick={() => {}}>
            <Profile.Picture
              src="/images/movie.png"
              data-testid="profile-picture"
            />
            <Profile.Name>Xiang Hou</Profile.Name>
          </Profile.User>
        </Profile.List>
      </Profile>
    );

    expect(getByText("Who's watching?")).toBeTruthy();
    expect(getByTestId("profile-picture")).toBeTruthy();
    expect(getByText("Xiang Hou")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it("renders <Profile /> with populated data but misc profile image", () => {
    const { container, getByText, getByTestId } = render(
      <Profile>
        <Profile.Title>Who's watching?</Profile.Title>
        <Profile.List>
          <Profile.User onClick={() => {}}>
            <Profile.Picture src="" data-testid="profile-picture-misc" />
            <Profile.Name>Xiang Hou</Profile.Name>
          </Profile.User>
        </Profile.List>
      </Profile>
    );

    expect(getByText("Who's watching?")).toBeTruthy();
    expect(getByTestId("profile-picture-misc")).toBeTruthy();
    expect(getByText("Xiang Hou")).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
