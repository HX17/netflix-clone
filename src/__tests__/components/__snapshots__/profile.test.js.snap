// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile /> renders <Profile /> with populated data but misc profile image 1`] = `
<div
  class="profile__Container-g30w1w-0 jFTqkz"
>
  <h1
    class="profile__Title-g30w1w-1 fA-DMvd"
  >
    Who's watching?
  </h1>
  <ul
    class="profile__List-g30w1w-2 dRnJLM"
  >
    <li
      class="profile__User-g30w1w-5 knXlGS"
    >
      <img
        class="profile__Picture-g30w1w-4 fERxgX"
        data-testid="profile-picture-misc"
        src="/images/misc/loading.gif"
      />
      <p
        class="profile__Name-g30w1w-3 hdArAc"
      >
        Xiang Hou
      </p>
    </li>
  </ul>
</div>
`;

exports[`<Profile /> renders the <Profile /> with populated data 1`] = `
<div
  class="profile__Container-g30w1w-0 jFTqkz"
>
  <h1
    class="profile__Title-g30w1w-1 fA-DMvd"
  >
    Who's watching?
  </h1>
  <ul
    class="profile__List-g30w1w-2 dRnJLM"
  >
    <li
      class="profile__User-g30w1w-5 knXlGS"
    >
      <img
        class="profile__Picture-g30w1w-4 fERxgX"
        data-testid="profile-picture"
        src="/images/users//images/movie.png.png"
      />
      <p
        class="profile__Name-g30w1w-3 hdArAc"
      >
        Xiang Hou
      </p>
    </li>
  </ul>
</div>
`;
